draft-ietf-quic-http.md

```diff
@@ -492,16 +492,32 @@ this header is determined by the stream type.
 ~~~~~~~~~~
 {: #fig-stream-header title="Unidirectional Stream Header"}
 
-Two stream types are defined in this document: control streams
-({{control-streams}}) and push streams ({{server-push}}).  Other stream types
-can be defined by extensions to HTTP/QUIC.
+Some stream types are reserved ({{stream-grease}}).  Two stream types are
+defined in this document: control streams ({{control-streams}}) and push streams
+({{server-push}}).  Other stream types can be defined by extensions to
+HTTP/QUIC.
 
 If the stream header indicates a stream type which is not supported by the
-recipient, this SHOULD be treated as a stream error of type
-HTTP_UNKNOWN_STREAM_TYPE.  The semantics of the remainder of the stream are
-unknown. Implementations SHOULD NOT send stream types the peer is not already
-known to support, since a stream error can be promoted to a connection error at
-the peer's discretion (see {{errors}}).
+recipient, the remainder of the stream cannot be consumed as the semantics are
+unknown. Recipients of unknown stream types MAY trigger a QUIC STOP_SENDING
+frame with an error code of HTTP_UNKNOWN_STREAM_TYPE, but MUST NOT consider such
+streams to be an error of any kind.
+
+Implementations SHOULD NOT send stream types the peer is not already known to
+support.   Stream types which could modify the state or semantics of existing
```

Comment:
This text recommends (SHOULD NOT) negotiating supported stream types. Is this recommendation necessary?

Reply:
The recommendation is purely because if you send types that aren't supported, you're wasting bandwidth.  It's not an interop issue, so perhaps that doesn't rise to the level of SHOULD NOT.  Would "MAY send" be more affirming?

Reply:
If you want to make that change it would make sense to add the sentence from this paragraph to the previous one e.g. 

_Implementations MAY send stream types the peer is not already known to support. If the recipient does not support a stream type, the remainder of the stream cannot be consumed as the semantics are unknown. (optional: add comment about wasted bandwidth)..._